CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/utilities/DateTimePicker.java

```diff
@@ -126,7 +121,15 @@ public LocalDate fromString(final String string) {
         mainGridPane.add(datePickerGridPane, 0, 0);
         mainGridPane.add(timePickerGrid, 0, 1);
         mainGridPane.setPadding(new Insets(1, 1, 1, 1));
+    }
 
+    private void validateInput(final String oldValue, final String newValue, final Spinner spinner) {
+        if (!newValue.matches("\\d*")) { // If the new typed text is not a number, revert to old text.
```

Comment:
can you declare this regex as a Pattern and compile beforehand as this will be more performant  (see sonar issue)